api/protos/aggregation/v1/aggregation.proto

```diff
@@ -0,0 +1,179 @@
+syntax = "proto3";
+
+package aggregation;
+option go_package = "aggregation/v1;aggregationv1";
+
+import "validate/validate.proto";
+
+
+// [#next-free-field: 2]
+message KeyerConfiguration {
+
+  // [#next-free-field: 2]
+  message Fragment {
+
+    // A rule defining how to match a Envoy request and what resulting
+    // fragment to generate.
+    // [#next-free-field: 3]
+    message Rule {
```

Comment:
@eapolinario qs for my understanding: what is the criteria for choosing when to keep a proto message within another message or outside?
I see that Rule is defined inside Fragment and others are outside.

Reply:
Majority of this from the spec so I'll shed some light on why I went about it this way. The ones that are inside the proto message are contained, similar to Java inner classes. They won't be used by other outer protos. The ones defined outside are shared.

Reply:
Thank you for explaining